domain/network/service/container_test.go

```diff
@@ -0,0 +1,261 @@
+	}}
+
+	spaceTwo := "two-space-uuid"
+	boundSpaces := []internal.SpaceName{{
+		UUID: spaceTwo,
+		Name: "two-space",
+	}}
+
+	negativeSpaces := []internal.SpaceName{{
+		UUID: "negative-space-uuid",
+		Name: "negative-space",
+	}}
+
+	exp := s.st.EXPECT()
+	exp.GetMachineSpaceConstraints(c.Context(), s.guestUUID.String()).Return(consSpaces, negativeSpaces, nil)
+	exp.GetMachineAppBindings(c.Context(), s.guestUUID.String()).Return(boundSpaces, nil)
+	exp.GetMachineNetNodeUUID(c.Context(), s.hostUUID.String()).Return(s.nodeUUID, nil)
+	// An ethernet device in the space can be bridged to satisfy one space
+	// requirement and an existing bridge satisfies the other.
+	exp.NICsInSpaces(c.Context(), s.nodeUUID, []string{spaceOne, spaceTwo}).Return(map[string][]network.NetInterface{
+		spaceOne: {{
+			Name:       "eth0",
+			Type:       corenetwork.EthernetDevice,
+			MACAddress: ptr("some-mac-address"),
+		}},
+		spaceTwo: {{
+			Name: "br-not-default-lxd",
+			Type: corenetwork.BridgeDevice,
+		}},
+	}, nil)
+	exp.GetContainerNetworkingMethod(c.Context()).Return(containermanager.NetworkingMethodProvider.String(), nil)
+
+	nics, err := s.svc.DevicesToBridge(c.Context(), s.hostUUID, s.guestUUID)
+	c.Assert(err, tc.ErrorIsNil)
+
+	c.Assert(nics, tc.HasLen, 1)
+	c.Check(nics[0], tc.DeepEquals, network.DeviceToBridge{
+		DeviceName: "eth0",
+		BridgeName: "br-eth0",
+		MACAddress: "some-mac-address",
+	})
+}
+
+func (s *containerSuite) TestDevicesToBridgeLocalBridgeReqsUnsatisfiable(c *tc.C) {
+	defer s.setupMocks(c).Finish()
+
+	s.setupServiceAndMachines(c)
+
+	spaceUUID := "positive-space-uuid"
+	spaces := []internal.SpaceName{{
+		UUID: spaceUUID,
+		Name: "positive-space",
+	}}
+
+	exp := s.st.EXPECT()
+	exp.GetMachineSpaceConstraints(c.Context(), s.guestUUID.String()).Return(spaces, nil, nil)
+	exp.GetMachineAppBindings(c.Context(), s.guestUUID.String()).Return(nil, nil)
+	exp.GetMachineNetNodeUUID(c.Context(), s.hostUUID.String()).Return(s.nodeUUID, nil)
+	// The default LXD bridge cannot satisfy space requirements when the
+	// container networking method is "provider".
+	exp.NICsInSpaces(c.Context(), s.nodeUUID, []string{spaceUUID}).Return(map[string][]network.NetInterface{
+		spaceUUID: {{
+			Name: internalnetwork.DefaultLXDBridge,
+			Type: corenetwork.BridgeDevice,
+		}},
+	}, nil)
+	exp.GetContainerNetworkingMethod(c.Context()).Return(containermanager.NetworkingMethodProvider.String(), nil)
+
+	_, err := s.svc.DevicesToBridge(c.Context(), s.hostUUID, s.guestUUID)
+	c.Assert(err, tc.ErrorIs, errors.SpaceRequirementsUnsatisfiable)
+}
+
+func (s *containerSuite) setupMocks(c *tc.C) *gomock.Controller {
+	ctrl := gomock.NewController(c)
+	s.st = NewMockState(ctrl)
+	c.Cleanup(func() { s.st = nil })
+	return ctrl
+}
+
+func (s *containerSuite) setupServiceAndMachines(c *tc.C) {
+	var err error
+
+	s.hostUUID, err = machine.NewUUID()
+	c.Assert(err, tc.ErrorIsNil)
+
+	s.guestUUID, err = machine.NewUUID()
+	c.Assert(err, tc.ErrorIsNil)
```

Comment:
```suggestion
	s.hostUUID = machinetesting.GenUUID(c)

	s.guestUUID = machinetesting.GenUUID(c)
```

Reply:
These don't actually pay their way. The spread of domains in which they are used mean they violate SRP, and it's a two line method, which is "shallow".